src/Symfony/Component/Notifier/Message/SmsMessage.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Message;
+
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Notification\Notification;
+use Symfony\Component\Notifier\Notification\SmsNotificationInterface;
+use Symfony\Component\Notifier\Receiver\Receiver;
+use Symfony\Component\Notifier\Receiver\SmsReceiverInterface;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 5.0
+ */
+final class SmsMessage implements MessageInterface
+{
+    private $transport;
+    private $subject;
+    private $phone;
+
+    public function __construct(string $phone, string $subject)
+    {
+        $this->subject = $subject;
+        $this->phone = $phone;
+    }
+
+    public static function fromNotification(Notification $notification, Receiver $receiver, string $transport = null): self
+    {
+        if (!$receiver instanceof SmsReceiverInterface) {
+            throw new LogicException(sprintf('To send a SMS message, "%s" should implement "%s" or the receiver should implement "%s".', get_class($notification), SmsNotificationInterface::class, SmsReceiverInterface::class));
+        }
+
+        return new self($receiver->getPhone(), $notification->getSubject());
+    }
+
+    /**
+     * @return $this
+     */
+    public function phone(string $phone)
+    {
+        $this->phone = $phone;
+
+        return $this;
+    }
+
+    public function getPhone(): string
+    {
+        return $this->phone;
+    }
+
+    public function getRecipientId(): ?string
+    {
+        return $this->phone;
+    }
+
+    /**
+     * @return $this
+     */
+    public function subject(string $subject)
```

Comment:
`: self`

Reply:
just a small question: why the set is omited? `subject()` and not `setSubject()`
for example in twig, we can use for a get purpose a code like this : object.propery => implies getter
here object.propery is a setter, it is a bit confusing

why not actual `getSubject` became `subject()` and actual `subject()` became `setSubject()` ?